tmail-backend/webadmin/webadmin-mailbox/src/main/java/com/linagora/tmail/webadmin/cleanup/CleanupTrashTask.java

```diff
@@ -0,0 +1,38 @@
+package com.linagora.tmail.webadmin.cleanup;
+
+import java.util.Optional;
+
+import org.apache.james.task.Task;
+import org.apache.james.task.TaskExecutionDetails;
+import org.apache.james.task.TaskType;
+
+public class CleanupTrashTask implements Task {
+    static final TaskType TASK_TYPE = TaskType.of("cleanup-trash");
+
+    private final CleanupTrashService cleanupTrashService;
+    private final RunningOptions runningOptions;
+
+    public CleanupTrashTask(CleanupTrashService cleanupTrashService, RunningOptions runningOptions) {
+        this.cleanupTrashService = cleanupTrashService;
+        this.runningOptions = runningOptions;
+    }
+
+    @Override
+    public Result run() throws InterruptedException {
+        return cleanupTrashService.cleanupTrash(runningOptions).block();
+    }
+
+    @Override
+    public TaskType type() {
+        return TASK_TYPE;
+    }
+
+    public RunningOptions getRunningOptions() {
+        return runningOptions;
+    }
+
+    @Override
+    public Optional<TaskExecutionDetails.AdditionalInformation> details() {
+        return Task.super.details();
```

Comment:
Should we return some useful information?

Reply:
what info?

Reply:
Whatever information you think is helpful, you can have a look at others AdditionalInformation implementation.

Reply:
@Arsnael should we add sth in AdditionalInformation?

Reply:
Number of deleted messages
Number of users processed

Anything else is relevant?

Reply:
Confirmed this above during daily